src/client/Microsoft.Identity.Client/MsalErrorMessage.cs

```diff
@@ -278,6 +278,10 @@ public static string RedirectUriMismatch(string expectedUri, string actualUri)
             " Potential issue: the redirect URI is valid, but it has been configured for the wrong app type." +
             " Check the configuration of the app being used in the app registration portal." +
             " See https://aka.ms/msal-net-invalid-client for details. ";
+        
+        internal const string ClientCredentialExplicitTenantId =
+            "You didn't configure an authority or you are trying to use an AAD authority with a general audience (`common`, `organizations` or `consumers`). " +
+            "Specify the tenant id with your authority. You can get tenant id from the app registration portal. Use a method like .WithAuthority(audience, tenantID) to specify the authority.";
```

Comment:
or maybe instead of (or in addition to) "You can get tenant id from the app registration portal. Use a method like .WithAuthority(audience, tenantID) to specify the authority.", add ". See https://aka.ms-msal-net-client-credentials" ?